File: Ahorn/triggers/BossFightTrigger.jl
module BossFightTrigger

using ..Ahorn, Maple

@mapdef Trigger "MaggyHelper/BossFightTrigger" BossFightTrigger(x::Integer, y::Integer, bossMusic::String="event:/music/lvl9/main", bossType::String="KirbyBoss", height::Integer=16, lockRoom::Bool=true, playMusic::Bool=true, width::Integer=16)

const placements = Ahorn.PlacementDict(
    "BossFightTrigger" => Ahorn.EntityPlacement(BossFightTrigger)
)

function Ahorn.selection(entity::BossFightTrigger)
    return Ahorn.getEntityRectangle(entity)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::BossFightTrigger, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.getEntityRectangle(entity), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/triggers/bossFightTrigger.lua
-- Loenn plugin for MaggyHelper - Boss Fight Trigger
local trigger = {}

trigger.name = "MaggyHelper/BossFightTrigger"
trigger.placements = {
    name = "BossFightTrigger",
    data = {
        width = 16,
        height = 16,
        bossType = "KirbyBoss",
        lockRoom = true,
        playMusic = true,
        bossMusic = "guid://{38e2f39c-382d-4136-86fd-e24520f3b71e}"
    }
}

trigger.fieldInformation = {
    bossType = {
        options = {
            "KirbyBoss",
            "DededeBoss",
            "MetaKnightBoss"
        },
        editable = false
    }
}

trigger.fieldOrder = {
    "x", "y", "width", "height",
    "bossType",
    "lockRoom",
    "playMusic",
    "bossMusic"
}

return trigger

